src/features/rewards/walletWrapper/index.tsx

```diff
@@ -318,6 +335,98 @@ export default class WalletWrapper extends React.PureComponent<Props, State> {
     )
   }
 
+  generateWalletButton = (walletState: WalletState) => {
+    const buttonProps: Partial<ButtonProps> = {
+      size: 'small',
+      level: 'primary',
+      brand: 'rewards',
+      type: (walletState === 'disconnected') ? 'subtle' : 'accent',
+      beforeIcon: (walletState !== 'unverified') ? <UpholdSystemIcon /> : undefined,
+      onClick: ['connected', 'verified'].includes(walletState)
+        ? this.toggleVerificationDetails
+        : this.props.onVerifyClick
+    }
+
+    switch (walletState) {
+      case 'unverified':
+        return (
+          <StyledWalletButton
+            text={getLocale('walletButtonUnverified')}
+            afterIcon={<StyledTextIcon>!</StyledTextIcon>}
+            {...buttonProps}
+          />
+        )
+
+      case 'connected':
+        return (
+          <StyledWalletButton
+            text={getLocale('walletButtonConnected')}
+            afterIcon={<StyledTextIcon>?</StyledTextIcon>}
+            {...buttonProps}
+          />
+        )
+
+      case 'verified':
+        return (
+          <StyledGreenWalletButton
+            text={getLocale('walletButtonVerified')}
+            afterIcon={<CaratDownIcon />}
+            {...buttonProps}
+          />
+        )
+
+      case 'disconnected':
+        return (
+          <Button
+            text={getLocale('walletButtonDisconnected')}
+            {...buttonProps}
+          />
+        )
+    }
+  }
+
+  toggleVerificationDetails = () => {
+    this.setState({ verificationDetails: !this.state.verificationDetails })
+  }
+
+  getVerificationDetails = (
+    verified: boolean,
+    username: string,
+    onVerifyClick: () => void,
+    goToUphold?: () => void
+  ) => {
+
+    return (
+      <WalletPopup
+        onClose={this.toggleVerificationDetails}
+        {...{ username, verified }}
+      >
+        {
+          verified
+            ? <StyledDialogList>
+              <li>
+                <StyledLink onClick={goToUphold} target={'_blank'}>
+                  {getLocale('walletGoToUphold')}
+                </StyledLink>
+              </li>
+              <li>
+                <StyledLink onClick={onVerifyClick}>
+                  {getLocale('walletReconnect')}
+                </StyledLink>
+              </li>
+            </StyledDialogList>
+            : <>
+              <StyledLink onClick={onVerifyClick}>
+                {getLocale('walletVerifyLink')}
+              </StyledLink>
+              {' '}
```

Comment:
Is this for stylistic purposes? Can we maybe accomplish this via CSS if so?

Reply:
It's not really for stylistic purposes, there should just be a space between the link and the rest of the test and the space will get eaten up by JSX otherwise.